Lib/test/test_capi/test_opt.py

```diff
@@ -1437,6 +1437,38 @@ def crash_addition():
 
         crash_addition()
 
+    def test_narrow_type_to_constant_str_empty(self):
+        def f(n):
+            trace = []
+            for i in range(n):
+                dummy = "aaa"
+                # Hopefully the optimizer can't guess what the value is.
+                # f is always "", but we can only prove that it's a string:
+                f = dummy[:0]
```

Comment:
I just realized this has the same name as the function it's in. Mind renaming it to `empty` or something? Also where it's used below.

Reply:
Fixed!